universe/node/nfts.go

```diff
@@ -102,6 +102,27 @@ func (n *Node) AddStakeActivities(stakeEvents []*harvester.StakeEvent) error {
 
 func (n *Node) AddStakeActivity(stakeEvent *harvester.StakeEvent) error {
 	node := universe.GetNode()
+
+	_, ok := n.GetObject(stakeEvent.OdysseyID, true)
+	if !ok {
+		// An event for an unknown object, in other words: the 'world' hasn't been created yet.
+		// Can easily happen during local development, when when DB and blockchain are out-of-sync.
+		// In prod could happen if people by bypass our UI/frontend, and directly stake on BC.
+		n.log.Errorf("Stake activity %s for unknown object %s", stakeEvent.TxHash, stakeEvent.OdysseyID)
+		// Can't really easily recover from this, getting stake activity before 'create' activity would be weird.
+		// Creating empty/unclaimed worlds is what we try to avoid by lazily creating worlds.
+		// For now, ingoring
+		return nil
+	}
+	aType := entry.ActivityTypeStake
+	if stakeEvent.ActivityType == "unstake" {
```

Comment:
Problem is, the field in stakeEvent is defined as "string", i suspect because it's JSON serialized somewhere...
So didn't wanna change too much (this bit of code was only moved up some lines) as this is a 'hotfix' for prod :/

Reply:
aw :/